### PR TITLE
add WalletLink connector

### PR DESCRIPTION
**Pull request by talentlessguy**
This PR adds a `WalletLink` connector to the alpha version of w3r. Connector code is very similar to MetaMask one. 

Preview:

image

### CODE REVIEW

File: packages/walletlink/package.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  "module": "dist/index.js",
+  "types": "dist/index.d.ts",
+  "scripts": {
+    "prebuild": "rm -rf dist",
+    "build": "tsc",
+    "start": "tsc --watch"
+  },
+  "dependencies": {
+    "@web3-react/types": "^8.0.2-alpha.0",
+    "walletlink": "^2.1.11"
```
Comment (NoahZinsmeister) [inline packages/walletlink/package.json:17]: can we make this a peer dep like in `@web3-react/walletconnect`? will also need to add this as a dependency to `example`. the motivation here is easier version upgrades for consumers

File: packages/walletlink/package.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "@web3-react/walletlink",
+  "publishConfig": {
+    "access": "public"
+  },
+  "version": "8.0.2-alpha.0",
+  "type": "module",
+  "module": "dist/index.js",
+  "types": "dist/index.d.ts",
```
Comment (NoahZinsmeister) [inline packages/walletlink/package.json:9]: update for consistency based on latest `main`
```suggestion
  "module": "./dist/index.js",
  "types": "./dist/index.d.ts",
```

File: packages/walletlink/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
+import { Connector, Actions, Provider } from '@web3-react/types'
+import type { WalletLinkOptions } from 'walletlink/dist/WalletLink'
```
Comment (NoahZinsmeister) [inline packages/walletlink/src/index.ts:2]: they don't export this from the top level package?

**Review by NoahZinsmeister — COMMENTED** [review]
LGTM after changes

File: packages/walletlink/package.json (hunk 1 above)
Comment (talentlessguy) [reply packages/walletlink/package.json:17]: sounds good, will change it now

Comment (talentlessguy) [reply packages/walletlink/package.json:19]: done

**Pull request merged**